internal/sink/sink.go

```diff
@@ -63,6 +65,10 @@ type SinkConfig struct {
 	NamespaceRefetch time.Duration
 	// OnFlushSuccess is an optional lifecycle hook
 	OnFlushSuccess func(string, int64)
+
+	// BatchHookOnFlushSuccess is an optional lifecycle hook if an external party wishes to
+	// performa additional operations on the batch of messages that were successfully flushed.
+	BatchHookOnFlushSuccess func([]SinkMessage)
```

Comment:
Another question. Where would the `BatchHookOnFlushSuccess` callback be invoked?

Reply:
I will make it so that it's executed in a background go thread. That should be fine in terms of non-blockingness.

This is used in the cloud implementation to meter the number of ingested events.